spec/controllers/training_modules_users_controller_spec.rb

```diff
@@ -30,6 +33,17 @@
           .to be_between(1.minute.ago, 1.minute.from_now)
       end
     end
-  end
+    context 'nonexistent slide' do
+      let(:nonexistent_slide_params) do
+        { module_id: t_mod.slug,
+          slide_id: 'not-a-real-slide',
+          user_id: user.id }
+      end
 
+      it 'renders a response' do
+        post :create_or_update, nonexistent_slide_params, format: :json
+        expect(response).to be
```

Comment:
Not sure. It's complicated by the fact that the React Router doesn't handle our 404s. That's why there's a `window.location` call in the React component. So here, it's hard to say what the test should do.